test/integration/eni-subnet-discovery/eni_subnet_discovery_test.go

```diff
@@ -0,0 +1,331 @@
+					checkSecondaryENISubnets(true)
+				})
+
+				Context("missing ec2:DescribeSubnets permission,", func() {
+					var role string
+					var EKSCNIPolicyV4ARN string
+					BeforeEach(func() {
+						By("getting the iam role")
+						podList, err := f.K8sResourceManagers.PodManager().GetPodsWithLabelSelector(AwsNodeLabelKey, utils.AwsNodeName)
+						Expect(err).ToNot(HaveOccurred())
+						for _, env := range podList.Items[0].Spec.Containers[0].Env {
+							if env.Name == "AWS_ROLE_ARN" {
+								role = strings.Split(env.Value, "/")[1]
+							}
+						}
+						if role == "" { // get the node instance role
+							By("getting the node instance role")
+							instanceProfileRoleName := strings.Split(*primaryInstance.IamInstanceProfile.Arn, "instance-profile/")[1]
+							instanceProfileOutput, err := f.CloudServices.IAM().GetInstanceProfile(instanceProfileRoleName)
+							Expect(err).ToNot(HaveOccurred())
+							role = *instanceProfileOutput.InstanceProfile.Roles[0].RoleName
+						}
+						err = f.CloudServices.IAM().DetachRolePolicy(EKSCNIPolicyARN, role)
+						Expect(err).ToNot(HaveOccurred())
+
+						eksCNIPolicyV4Path := utils.GetProjectRoot() + EKSCNIPolicyV4
+						eksCNIPolicyV4Bytes, err := os.ReadFile(eksCNIPolicyV4Path)
+						Expect(err).ToNot(HaveOccurred())
+
+						eksCNIPolicyV4Data := string(eksCNIPolicyV4Bytes)
+
+						By("Creating and attaching policy AmazonEKS_CNI_Policy_V4")
+						output, err := f.CloudServices.IAM().CreatePolicy("AmazonEKS_CNI_Policy_V4", eksCNIPolicyV4Data)
+						Expect(err).ToNot(HaveOccurred())
+						EKSCNIPolicyV4ARN = *output.Policy.Arn
+						err = f.CloudServices.IAM().AttachRolePolicy(EKSCNIPolicyV4ARN, role)
+						Expect(err).ToNot(HaveOccurred())
+
+						// Sleep to allow time for CNI policy reattachment
+						time.Sleep(10 * time.Second)
+
+						RestartAwsNodePods()
+					})
+
+					AfterEach(func() {
+						By("attaching VPC_CNI policy")
+						err = f.CloudServices.IAM().AttachRolePolicy(EKSCNIPolicyARN, role)
+						Expect(err).ToNot(HaveOccurred())
+
+						By("Detaching and deleting policy AmazonEKS_CNI_Policy_V4")
+						err = f.CloudServices.IAM().DetachRolePolicy(EKSCNIPolicyV4ARN, role)
+						Expect(err).ToNot(HaveOccurred())
+
+						err = f.CloudServices.IAM().DeletePolicy(EKSCNIPolicyV4ARN)
+						Expect(err).ToNot(HaveOccurred())
+
+						// Sleep to allow time for CNI policy detachment
+						time.Sleep(10 * time.Second)
+
+						RestartAwsNodePods()
+					})
+					It("should have same subnet as primary ENI", func() {
+						checkSecondaryENISubnets(false)
+					})
+				})
+			})
+			Context("using a subnet tagged with kubernetes.io/role/cn", func() {
+				BeforeEach(func() {
+					By("Tagging kubernetes.io/role/cn to subnet")
+					_, err = f.CloudServices.EC2().
+						CreateTags(
+							[]string{createdSubnet},
+							[]*ec2.Tag{
+								{
+									Key:   aws.String("kubernetes.io/role/cn"),
```

Comment:
`kubernetes.io/role/cni`instead of `cn`?

Reply:
I think, it is alright. The tagging and un-tagging was consistent in this test.